### PR TITLE
[clang][fmv] Drop .ifunc from target_version's entrypoint's mangling

### LINKED ISSUE #81043: [FMV] mangling of forward-declared multi-versioned callee differs between gcc and clang
https://clang.godbolt.org/z/zeeTxd3va

```
__attribute__((target_version("fp16")))
void foo(void);

__attribute__((target_version("fp16fml")))
void foo(void);

__attribute__((target_version("default")))
void foo(void);

void caller() {
    foo();
}
```

Note how clang calls it `_Z3foov.ifunc`, and gcc refers to it as `_Z3foov`. IMO gcc's behavior is the desirable one: it lets you forward-declare foo in a header without having to promise that the implementation is multi-versioned. This is useful for libraries that have to provide a stable ABI to be able to adopt FMV in the implementation of those interfaces, without having to write an excessive number of shims.

### PR DESCRIPTION
**Pull request by jroelofs**
Fixes: https://github.com/llvm/llvm-project/issues/81043

### CODE REVIEW

File: clang/lib/CodeGen/CodeGenModule.cpp [hunk 1]
```diff
@@ -4393,7 +4394,7 @@ llvm::Constant *CodeGenModule::GetOrCreateMultiVersionResolver(GlobalDecl GD) {
   // a separate resolver).
   std::string ResolverName = MangledName;
   if (getTarget().supportsIFunc()) {
-    if (!FD->isTargetClonesMultiVersion())
+    if (!FD->isTargetClonesMultiVersion() && !FD->isTargetVersionMultiVersion())
```
[the remark this answers is not in the document]
Comment (jroelofs) [reply clang/lib/CodeGen/CodeGenModule.cpp:4397]: `target` and the two cpu-specific ones. I'll change it to a switch.

[the remark this answers is not in the document]
Comment (elizabethandrews) [reply clang/lib/CodeGen/CodeGenModule.cpp:4397]: I didn't realize there were multiple attributes still using this mangling. A switch would make this code better but I'm ok either way.

**Comment by labrinea** [comment]
> Looks like we've done something similar for `target_clones` to fix a linker issue: [cee5b87](https://github.com/llvm/llvm-project/commit/cee5b8777fa98312b05bf8aa81554910a8f867c5). What is the motive in this case?

Ah, I just saw that it fixes https://github.com/llvm/llvm-project/issues/81043.

**Review by labrinea — APPROVED** [review]
LGTM

**Review by DanielKristofKiss — APPROVED** [review]
Thanks!

**Pull request merged**